### PR TITLE
Bootstrap - setup project basis.

### PR DESCRIPTION
**Pull request by Darmody**
Setup react & redux & babel & webpack & eslint & mocha enviroment!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules/
```
Comment (kreja) [inline .gitignore:1]: You should adding `.DS_STORE`

File: package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    "development": "node ./webpack-dev-server.js",
+    "test": "NODE_ENV=test mocha --recursive --compilers js:babel-core/register --require ./test/setup.js",
+    "lint": "eslint -c .eslintrc --fix app"
+  },
+  "betterScripts": {},
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/Darmody/Gecko.git"
+  },
+  "keywords": [],
```
Comment (kreja) [inline package.json]: you should add some keywords, like gecko or something

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    'webpack-hot-middleware/client',
+    './app'
+  ],
+  output: {
+    path: path.join(__dirname, 'bundle'),
+    filename: 'app.js',
+    publicPath: '/bundle/'
+  },
+  plugins: [
+    new webpack.optimize.OccurenceOrderPlugin(),
```
Comment (kreja) [inline webpack.config.js:16]: Why not make the `preferEntry` true? I haven't used this plugin, so I'm not what the effects will be, but the document recommends to do so.
[occurenceorderplugin](http://webpack.github.io/docs/list-of-plugins.html#occurenceorderplugin)

File: .gitignore (hunk 1 above)
Comment (Darmody) [inline .gitignore:1]: u are right.

File: package.json (hunk 2 above)
Comment (Darmody) [inline package.json]: it's a development project now. How about just add gecko for the time being?

File: webpack.config.js (hunk 3 above)
Comment (Darmody) [inline webpack.config.js:16]: I dont know about this feature too. I think we can test this feature during developing.

Comment (kreja) [inline webpack.config.js:16]: ok~

File: package.json (hunk 2 above)
Comment (kreja) [inline package.json]: ok.. this comment is too difficult to find...

**Comment by kreja** [comment]
:+1: 
:cupid:

**Pull request merged**